framework/db/Connection.php

```diff
@@ -607,6 +611,7 @@ protected function initConnection()
         if ($this->charset !== null && in_array($this->getDriverName(), ['pgsql', 'mysql', 'mysqli', 'cubrid'])) {
             $this->pdo->exec('SET NAMES ' . $this->pdo->quote($this->charset));
         }
+        $this->getSupportsResultSets();
```

Comment:
Then initialization could be done here once and `getSupportsResultSets()` could return the value that's stored.